src/main/java/org/spongepowered/server/mixin/entity/player/MixinEntityPlayer.java

```diff
@@ -110,6 +119,16 @@ public void onWriteEntityToNBT(NBTTagCompound tagCompound, CallbackInfo ci) {
         tagCompound.setTag("Spawns", spawnlist);
     }
 
+    @Inject(method = "interactWith", at = @At(value = "INVOKE_ASSIGN", target = "net/minecraft/entity/player/EntityPlayer"
+            + ".getCurrentEquippedItem()Lnet/minecraft/item/ItemStack;"), cancellable = true)
+    public void onInteractWith(net.minecraft.entity.Entity entity, CallbackInfoReturnable<Boolean> cir) {
+        InteractEntityEvent.Secondary event = SpongeEventFactory.createInteractEntityEventSecondary(SpongeImpl.getGame(),
+                Cause.of(NamedCause.source((Player) this)), Optional.empty(), (Entity) entity);
```

Comment:
There's no need to cast to `Player` here - `NamedCause.source` takes an `Object`